pkg/agent/util/net_windows.go

```diff
@@ -32,36 +34,74 @@ import (
+	commandRetryInternal = time.Second
 )
 
 func GetNSPath(containerNetNS string) (string, error) {
 	return containerNetNS, nil
 }
 
+func GetHostInterfaceStatus(ifaceName string) (string, error) {
+	cmd := fmt.Sprintf(`Get-NetAdapter -InterfaceAlias "%s" | Select-Object -Property Status | Format-Table -HideTableHeaders`, ifaceName)
+	out, err := CallPSCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(out), nil
+}
+
 // EnableHostInterface sets the specified interface status as UP.
 func EnableHostInterface(ifaceName string) error {
-	cmd := fmt.Sprintf("Enable-NetAdapter -InterfaceAlias %s", ifaceName)
-	return InvokePSCommand(cmd)
+	cmd := fmt.Sprintf(`Enable-NetAdapter -InterfaceAlias "%s"`, ifaceName)
+	// Enable-NetAdapter is not a block operation by our test. It returns immediately no matter if the interface
+	// has been enabled. So we need to check interface status to ensure the interface is up before return.
+	return wait.PollImmediate(commandRetryInternal, commandMaxRetry, func() (done bool, err error) {
+		if err := InvokePSCommand(cmd); err != nil {
+			klog.Errorf("Failed to run command %s: %v", cmd, err)
+			return false, nil
+		}
+		status, err := GetHostInterfaceStatus(ifaceName)
+		if err != nil {
+			klog.Errorf("Failed to run command %s: %v", cmd, err)
+			return false, nil
+		}
+		if !strings.EqualFold(status, "Up") {
+			klog.Errorf("Waiting for host interface %s to be up", ifaceName)
+			return false, nil
+		}
+		return true, nil
+	})
 }
 
 // ConfigureInterfaceAddress adds IPAddress on the specified interface.
 func ConfigureInterfaceAddress(ifaceName string, ipConfig *net.IPNet) error {
 	ipStr := strings.Split(ipConfig.String(), "/")
-	cmd := fmt.Sprintf("New-NetIPAddress -InterfaceAlias %s -IPAddress %s -PrefixLength %s", ifaceName, ipStr[0], ipStr[1])
-	return InvokePSCommand(cmd)
+	cmd := fmt.Sprintf(`New-NetIPAddress -InterfaceAlias "%s" -IPAddress %s -PrefixLength %s`, ifaceName, ipStr[0], ipStr[1])
+	err := InvokePSCommand(cmd)
+	// If the address already exists, ignore the error.
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return err
+	}
+	return nil
 }
 
 // ConfigureInterfaceAddressWithDefaultGateway adds IPAddress on the specified interface and sets the default gateway
 // for the host.
 func ConfigureInterfaceAddressWithDefaultGateway(ifaceName string, ipConfig *net.IPNet, gateway string) error {
 	ipStr := strings.Split(ipConfig.String(), "/")
-	cmd := fmt.Sprintf("New-NetIPAddress -InterfaceAlias %s -IPAddress %s -PrefixLength %s -DefaultGateway %s", ifaceName, ipStr[0], ipStr[1], gateway)
-	return InvokePSCommand(cmd)
+	cmd := fmt.Sprintf(`New-NetIPAddress -InterfaceAlias "%s" -IPAddress %s -PrefixLength %s -DefaultGateway %s`, ifaceName, ipStr[0], ipStr[1], gateway)
+	err := InvokePSCommand(cmd)
+	// If the address already exists, ignore the error.
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return err
+	}
+	return err
```

Comment:
I think it should return nil, otherwise "already exists" error will still fail the function.

Reply:
Thanks Quan. Good catch.

Reply:
Done.